### PR TITLE
Use realpath for path comparison in Makefile

### PR DESCRIPTION
**Pull request by YorikSar**
**What this PR does / why we need it**:

This prevents make from erroring if some part of `$(GOPATH)/src` is a symlink and `$(PWD)` might already be canonical.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -29,7 +29,7 @@ TAGS      :=
 LDFLAGS   := "-w -s -X 'main.version=${VERSION}'"
 REGISTRY ?= k8scloudprovider
 
-ifneq ("$(DEST)", "$(PWD)")
+ifneq ("$(realpath $(DEST))", "$(realpath $(PWD))")
```
Comment (chaosaffe) [inline Makefile]: `realpath` is not available on MacOS by default. Please add checks to ensure that `realpath` exists before using, and put a fall back in place if `realpath` is not available.

Comment (YorikSar) [reply Makefile]: I've checked that it works with `GNU Make 3.81` that is installed by default on MacOS (or rather with DevTools). `realpath` here is not an external binary, it's an internal Make command.

**Comment by YorikSar** [comment]
@chaosaffe I personally didn't run into symlink incompatibilities with Go tooling, at least in my setup. Only such explicit check would fail for me.

**Comment by chaosaffe** [comment]
xref: https://github.com/golang/go/issues/15507#issuecomment-241581036
/assign @chrigl

**Comment by chrigl** [comment]
@chaosaffe, I would give a LGTM. Might safe some debugging for some folks.
```
[12:18:28] ~/P/k/s/s/cluster-api-provider-openstack (0)  (master|…21)
> pwd
/Users/chris/Projects/kube-gopath/src/sigs.k8s.io.testing/cluster-api-provider-openstack
[12:18:43] ~/P/k/s/s/cluster-api-provider-openstack (0)  (master|…21)
> ls -l /Users/chris/Projects/kube-gopath/src/sigs.k8s.io
lrwxr-xr-x 1 chris staff 20 Dez 11 12:06 /Users/chris/Projects/kube-gopath/src/sigs.k8s.io -> sigs.k8s.io.testing/
[12:18:44] ~/P/k/s/s/cluster-api-provider-openstack (0)  (master|…21)
> make clusterctl
Makefile:33: *** Please run 'make' from /Users/chris/Projects/kube-gopath/src/sigs.k8s.io/cluster-api-provider-openstack. Current directory is /Users/chris/Projects/kube-gopath/src/sigs.k8s.io.testing/cluster-api-provider-openstack.  Stop.
```

While with `relpath` build works fine.
```
> git checkout YorikSar/makefile-realpath
Note: checking out 'YorikSar/makefile-realpath'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at d8c2c422 Use realpath for path comparison in Makefile
[12:19:42] ~/P/k/s/s/cluster-api-provider-openstack (0)  ((d8c2c422…)|…22)
> make clusterctl
CGO_ENABLED=0 GOOS=darwin go build \
                -ldflags "-w -s -X 'main.version=d8c2c422'" \
                -o bin/clusterctl \
                cmd/clusterctl/main.go
```

**Pull request merged**